core-backend/common/src/funcs.rs

```diff
@@ -432,9 +454,15 @@ where
     #[host(cost = RuntimeCosts::Debug(data_len))]
     pub fn debug(ctx: &mut R, data_ptr: u32, data_len: u32) -> Result<(), R::Error> {
         let read_data = ctx.register_read(data_ptr, data_len);
-        let data: RuntimeBuffer = ctx.read(read_data)?.try_into()?;
-
-        let s = String::from_utf8(data.into_vec())?;
+        let data: RuntimeBuffer = ctx
+            .read(read_data)?
+            .try_into()
+            .map_err(|RuntimeBufferSizeError| InfallibleMemoryError::RuntimeAllocOutOfBounds.into())
+            .map_err(TrapExplanation::InfallibleExt)?;
+
+        let s = String::from_utf8(data.into_vec())
+            .map_err(|_err| InfallibleExecutionError::InvalidDebugString.into())
```

Comment:
double error mapping seems strange

Reply:
Or it will be very terrible formatted code